src/ui/space-ui.vue

```diff
@@ -1,16 +1,254 @@
+      padding: 14px;
+      margin: 0;
+    }
+  }
+</style>
+
 <template>
-  <div>
-    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  <div class="spaceUI">
+    <div class="spaceUI-desc">
+        {{ spaceDescription}}
+    </div>
+    <div class="spaceUI-readout">
+      <ul class="spaceUI-list" ref="list">
+        <li v-for="(item, index) in spaceContent" class="spaceUI-list-item" :data-item-id="item.logicRef">
+          <button v-html="drag"></button>
+          <span class="spaceUI-list-item-name">{{ componentName(item) }} -- {{index}}</span>
+          <button v-html="target" v-on:click="openTarget(item.logicRef)"></button>
+          <button v-html="remove" v-on:click="removeFromSpace(item.logicRef)"></button>
+        </li>
+      </ul>
+      <button type="button" v-on:click="addComponent">Add Component To Space</button>
+      <button type="button" v-on:click="render">Render</button>
+    </div>
   </div>
 </template>
 
 <script>
+import { map, assign, filter, findIndex } from 'lodash';
+import { remove, removeSpace } from '../services/remove-service';
+import { openAddComponent } from '../services/ui-service';
+import targetIcon from '../../media/target.svg';
+import removeIcon from '../../media/remove.svg';
+import dragIcon from '../../media/drag.svg';
+import dragula from 'dragula';
+import { findAvailableComponents, addToSpace } from '../services/add-service';
+
+const utils = window.kiln.utils;
+
+
+// Placeholder for Dragula instance
+var drag;
+
+/**
+ * get index of a child element in a container
+ * @param {Element} el
+ * @param {Element} container
+ * @returns {number}
+ */
+function getIndex(el, container) {
+  return findIndex(container.children, (child) => child === el);
+}
+
+/**
+ * Add Dragula functionality
+ *
+ * @param {Element} el
+ * @param {Function} reorder
+ */
+function addDragula(el, reorder) {
+  var oldIndex;
+
+  drag = dragula([el], {
+    direction: 'vertical'
+  });
+
+  drag.on('drag', function (selectedItem, container) {
+    oldIndex = getIndex(selectedItem, container);
+  });
+
+  drag.on('cancel', function () {
+    oldIndex = null;
+  });
+
+  drag.on('drop', function (selectedItem, container) {
+    reorder(selectedItem.getAttribute('data-item-id'), getIndex(selectedItem, container), oldIndex, selectedItem);
+  });
+}
 
 export default {
   data() {
-    return {}
+    const state = this.$store.state,
+      spaceRef = state.ui.currentPane.content.spaceRef;
+
+    return {
+      spaceName: state.ui.currentPane.content.spaceName,
+      spaceRef,
+      items: state.components[spaceRef].content
```

Comment:
@jonwinton copying state from the store to component data means that you have to manually keep the two in sync (that's what `updateItems` does). if we use a computed property instead, we can get rid of updateItems and the three promise chains that call it.